src/main/java/org/jenkinsci/plugins/trileadapi/TrileadSSHPasswordAuthenticator.java

```diff
@@ -0,0 +1,197 @@
+                    return true;
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace(getListener().error("Failed to authenticate"));
+        }
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected boolean doAuthenticate() {
+        final StandardUsernamePasswordCredentials user = getUser();
+        final String username = getUsername();
+
+        try {
+            final Connection connection = getConnection();
+            final String password = user.getPassword().getPlainText();
+            boolean tried = false;
+
+            List<String> availableMethods = Arrays.asList(connection.getRemainingAuthMethods(username));
+            if (availableMethods.contains(PASSWORD)) {
+                // prefer password
+                if (connection.authenticateWithPassword(username, password)) {
+                    LOGGER.fine("Authentication with 'password' succeeded.");
+                    return true;
+                }
+                getListener().error("Failed to authenticate as %s. Wrong password. (credentialId:%s/method:password)",
+                        username, user.getId());
+                tried = true;
+            }
+            if (availableMethods.contains(KEYBOARD_INTERACTIVE)) {
+                if (connection.authenticateWithKeyboardInteractive(username, (name, instruction, numPrompts, prompt, echo) -> {
+                    // most SSH servers just use keyboard interactive to prompt for the password
+                    // match "assword" is safer than "password"... you don't *want* to know why!
+                    return prompt != null && prompt.length > 0 && prompt[0].toLowerCase(Locale.ENGLISH)
+                            .contains("assword")
+                            ? new String[]{password}
+                            : new String[0];
+                })) {
+                    LOGGER.fine("Authentication with  'keyboard-interactive' succeeded.");
+                    return true;
+                }
+                getListener()
+                        .error("Failed to authenticate as %s. Wrong password. "
+                                + "(credentialId:%s/method:keyboard-interactive)",
+                                username, user.getId());
+                tried = true;
+            }
+
+            if (!tried) {
+                getListener().error("The server does not allow password authentication. Available options are %s",
+                        availableMethods);
+            }
+        } catch (IOException e) {
+            e.printStackTrace(getListener()
+                    .error("Unexpected error while trying to authenticate as %s with credential=%s", username,
+                            user.getId()));
+        }
+        return false;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @OptionalExtension(requirePlugins = {"ssh-credentials"})
```

Comment:
this uses `StandardUsernamePasswordCredentials` not SSH ones, so the following?

```suggestion
    @OptionalExtension(requirePlugins = {"plain-credentials"})
```

Reply:
`StandardUsernamePasswordCredentials` is defined in `credentials`. `plain-credentials` is for string & text credentials which are newer.

Reply:
```suggestion
    @OptionalExtension(requirePlugins = {"credentials"})
```

Reply:
@jtnord `ssh-credentials` is required since this is a `com.cloudbees.jenkins.plugins.sshcredentials.SSHAuthenticator` implementation. `ssh-credentials` should be sufficient since it has a direct required dependency on `credentials` ? Or do we want to include both `ssh-credentials` and `credentials` ?